### PR TITLE
WIP: pastebin: new destination

### PR DESCRIPTION
**Pull request by Kokan**
I've been working on a way to store my logs in a cheap storage, where I could easily access without any hassle.

The idea just came today that https://pastebin.com provides us storage, and could store any textual data which the logs are. So here we go, please enjoy.

Example configuration:
```
pastebin(
  api-dev-key("<prive key from your https://pastebin.com/api page after login")
  expire-date("10M")
)
```
Further development could include the use of authenticating header, so the logs could be saved under a user and not as anonime.

### CODE REVIEW

File: scl/pastebin/plugin.conf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# For details see the pastebin API doc: https://pastebin.com/api
+
+block destination pastebin(
+  api-dev-key()
+  private(0)
+  expire-date("10M")
+  ...)
+{
+
+@requires http "The elasticsearch-http() driver depends on the syslog-ng http module, please install the syslog-ng-mod-http (Debian & derivatives) or the syslog-ng-http (RHEL & co) package"
```
Comment (alltilla) [inline scl/pastebin/plugin.conf:32]: copy-`paste` error ;)

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by alltilla** [comment]
It would be great to see the other optional parameters as well:
1. api_user_key - this paramater is part of the login system, which is explained further down the page.
2. api_paste_name - this will be the name / title of your paste.
3. api_paste_format - this will be the syntax highlighting value, which is explained in detail further down the page.

Sorry, I know, it is in WIP state, but couldn't contain myself to review this, as I find this really cool :smile:

**Comment by Kokan** [comment]
@alltilla As my current understanding the `api_user_key` is a secret that you could get like any authentication header stuff, so my last comment actually refers to getting these values and properly setting without user interaction.

The `api_paste_name` is cool, but probably it would be better with the support of appending the existing paste and not creating a new per message :)

This could do easily: `api_paste_format` without much of an effort.

**Comment by Kokan** [comment]
Saving this one for a rainy day: https://pastebin.com/1Zf4njbP

**Comment by faxm0dem** [comment]
I laughed !
This reminded me of this https://github.com/scrottie/CHI-Driver-Ping/blob/master/README (not an April joke)

**Comment by Kokan** [comment]
@faxm0dem People tried: https://code.kryo.se/pingfs/ . Actually I am working on a project where it is  a goal to store data in network via packets.

**Pull request closed without merging**